programmers/음양-더하기.js

```diff
@@ -0,0 +1,31 @@
+// reduce 사용.
+//
+function solution(absolutes, signs) {
+    const reducer = (previousValue, currentValue) => previousValue + currentValue;
+    
+    for (let i = 0; i < absolutes.length; i += 1) {
+        signs[i] === false ? 
+            absolutes.push(-absolutes[i] * 2)
+            :
+            absolutes[i] = absolutes[i];
+    }
+    return absolutes.reduce(reducer);
+}
+
+
+// for문 사용.
+// 
+// function solution(absolutes, signs) {
+// let result = 0;
+//     for (let i = 0; i < absolutes.length; i += 1) {
+//         signs[i] === false ? 
+//             absolutes.push(-absolutes[i] * 2)
+//             :
+//             absolutes[i] = absolutes[i];
+//     }
+    
+//     for (let i = 0; i < absolutes.length; i += 1) {
+//             result += absolutes[i];
+//         }
+//     return result;
+// }
```

Comment:
```suggestion
function solution(absolutes, signs) {
  let result = 0;
  
  for (let i = 0; i < absolutes.length; i += 1) {
    signs[i] ? (result += absolutes[i]) : (result -= absolutes[i]);
  }
  return result;
}
```
for문을 1회만 사용하고 해결할 수 있어요